src/test/java/com.gigssandbox/UserServiceTest.java

```diff
@@ -39,45 +39,45 @@ public void shoudAddUserToUsersCollectionWhenUserTriesToRegister() {
     @Test
     public void shouldThrowAlreadyRegisteredExceptionWhenUserCredentialsAreAlreadyPresentInUsersMap() {
         String username = "skip";
-        char[] password = "the foreplay".toCharArray();
-        Whitebox.setInternalState(userService, "users", Collections.singletonMap(username, User.builder().username(username).passwordHash(Arrays.hashCode(password)).build()));
+        String password = "the foreplay";
+        Whitebox.setInternalState(userService, "users", Collections.singletonMap(username, User.builder().username(username).passwordHash(password.hashCode()).build()));
 
         assertThrows(AlreadyRegisteredException.class, () -> userService.registerUser(username, password));
     }
 
     @Test
     public void shouldThrowUserNotRegisteredExceptionWhenUserCredentialsAreNotPresentInUsersMap() {
```

Comment:
1) you violate UserService internal structure with "WhenUserCredentialsAreNotPresentInUsersMap" where you point that UserService use Map. What if you change implementation?
2) just shouldThrow, because Exception type is obtained from annotation

When you write test and its name you should behave like you are client of him:

shouldThrowWhenUserIsAlreadyRegistered


same for every test